torch/_dynamo/side_effects.py

```diff
@@ -593,11 +593,22 @@ def codegen_update_mutated(self, cg: PyCodegen):
             elif isinstance(
                 var, variables.torch_function.TorchFunctionModeStackVariable
             ):
+                # Needed in the finally block for stack restoration
+                cg.add_push_null(
+                    lambda: cg.load_import_from(
+                        utils.__name__, "get_torch_function_mode_stack"
+                    )
+                )
+                cg.call_function(0, False)
+                name = variables.torch_function.get_prev_stack_var_name()
```

Comment:
Should this name be different for each `TorchFunctionModeVariable`? If there are multiple `TorchFunctionModeVariable` contexts on the stack, we would be referring to the same object.

Reply:
No, there's only one TorchFunctionModeStackVariable (it's a singleton). This is used to flag in side effects if there is any mutation of the stack. If there is, we generate the bytecode to update it accordingly. For the exception handling, we'll only ever restore the previous version of stack once, so it is only stashed in the local once.